test/integ/axpy/src/axpy.cpp

```diff
@@ -30,25 +30,25 @@ class AxpyKernel
     //! \tparam TAcc The type of the accelerator the kernel is executed on..
     //! \tparam TElem The matrix element type.
     //! \param acc The accelerator the kernel is executed on.
-    //! \param numElements Specifies the number of elements of the vectors X and Y.
     //! \param alpha Scalar the X vector is multiplied with.
     //! \param X Vector of at least n elements.
     //! \param Y Vector of at least n elements.
     ALPAKA_NO_HOST_ACC_WARNING
-    template<typename TAcc, typename TElem, typename TIdx>
+    template<typename TAcc, typename TMemoryHandleX, typename TMemoryHandleY, typename TElem, typename TIdx>
     ALPAKA_FN_ACC auto operator()(
         TAcc const& acc,
-        TIdx const& numElements,
         TElem const& alpha,
-        TElem const* const X,
-        TElem* const Y) const -> void
+        alpaka::experimental::Accessor<TMemoryHandleX, TElem, TIdx, 1, alpaka::experimental::ReadAccess> const X,
+        alpaka::experimental::Accessor<TMemoryHandleY, TElem, TIdx, 1, alpaka::experimental::ReadWriteAccess> const Y)
+        const -> void
     {
         static_assert(alpaka::Dim<TAcc>::value == 1, "The AxpyKernel expects 1-dimensional indices!");
 
         auto const gridThreadIdx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u];
         auto const threadElemExtent = alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u];
         auto const threadFirstElemIdx = gridThreadIdx * threadElemExtent;
 
+        auto const numElements = X.extents[0];
```

Comment:
The `X.extents[0]` isn't part of the `Accessor` interface (it only belongs to `ViewAccessor`).

Reply:
I wonder whether this should be obtainable through `getExtent`, too.

Reply:
IIRC, we don't have any other accessor than a ViewAccessor at the moment. And yes, we could think about accessors implementing `getExtent`. For now I went with the easier solution.

Reply:
Well, there are my SYCL accessors ;-)